pkg/storegateway/bucket_stores.go

```diff
@@ -464,7 +464,7 @@ func (u *BucketStores) getOrCreateStore(userID string) (*BucketStore, error) {
 		fetcher,
 		u.syncDirForUser(userID),
 		newChunksLimiterFactory(u.limits, userID),
-		NewSeriesLimiterFactory(0), // No series limiter.
+		newSeriesLimiterFactory(u.limits, userID), // No series limiter.
```

Comment:
The replaced `NewSeriesLimiterFactory` seems to now be only used in tests. Should we move it to test files? Same for `NewChunksLimiterFactory`

Reply:
Hi @dimitarvdimitrov, 
What if we modify `Limiter.Reserve` in such a way that, in case of error, it  returns `httpgrpc.Errorf(http.StatusUnprocessableEntity, "limit %v exceeded", l.limit)` instead of `errors.Errorf("limit %v exceeded", l.limit)`, and then use `Limiter.NewChunksLimiterFactory` and `Limiter.NewSeriesLimiterFactory` instead of the 2 methods used in `bucket_stores.go`? This way all limiter-related things could be moved away from `bucket_stores.go`?

Reply:
Yes, I think this sounds like a good idea. Not sure how much changes would be required to the existing tests, but SGTM